From b08cf37d64a57998085673608bc53bb831f83271 Mon Sep 17 00:00:00 2001
From: "Suresh, Prashanth" <prashanth.suresh@intel.com>
Date: Mon, 26 Sep 2022 15:20:26 +0530
Subject: [PATCH] Pass host graphics SRIOV info to guest.

Add file in aaf folder to enable guest to know if host supports SRIOV.
Set media coded to hardware based on this value.

Tracked-On: OAM-104076
Signed-off-by: Suresh, Prashanth <prashanth.suresh@intel.com>
---
 groups/media/auto/auto_hal.in | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/groups/media/auto/auto_hal.in b/groups/media/auto/auto_hal.in
index 7d0fbb5..4e68aec 100644
--- a/groups/media/auto/auto_hal.in
+++ b/groups/media/auto/auto_hal.in
@@ -13,13 +13,16 @@ case "$(cat /proc/fb)" in
                 setprop vendor.intel.video.codec hardware
                 ;;
         *)
-                if [ -d "/sys/class/drm/card0/engine" ];then
-                       echo "Intel SRIOV"
-                       setprop vendor.intel.video.codec hardware
-                else
-                       echo "software codec"
-                       setprop vendor.intel.video.codec software
-                fi
+                case "$(cat /mnt/share/mixins.spec)" in
+                        *gpu-type:sriov*)
+	                        echo "Intel SRIOV"
+                                setprop vendor.intel.video.codec hardware
+                                ;;
+                        *)
+                                echo "Software codec"
+                                setprop vendor.intel.video.codec software
+                                ;;
+                esac
                 ;;
 esac
 }
-- 
2.39.1

